planning/behavior_path_planner/src/marker_utils/utils.cpp

```diff
@@ -463,4 +487,162 @@ MarkerArray createPredictedPathMarkerArray(
+      marker_array.markers.emplace_back();
+      auto & polygon_marker = marker_array.markers.back();
+      polygon_marker = default_line_marker(color);
+      polygon_marker.points.reserve(polygon.outer().size());
+      for (const auto & p : polygon.outer()) {
+        polygon_marker.points.push_back(tier4_autoware_utils::createPoint(p.x(), p.y(), poly_z));
+      }
+    };
+
+    insert_polygon_marker(info.extended_ego_polygon);
+    insert_polygon_marker(info.extended_obj_polygon);
+
+    const auto str_idx = std::to_string(++idx);
+    const auto insert_text_marker = [&](const auto & pose) {
+      marker_array.markers.emplace_back();
+      auto & text_marker = marker_array.markers.back();
+      text_marker = default_text_marker();
+      text_marker.text = str_idx;
+      text_marker.pose = pose;
+    };
+
+    insert_text_marker(info.expected_ego_pose);
+    insert_text_marker(info.expected_obj_pose);
+
+    const auto insert_cube_marker = [&](const auto & pose) {
+      marker_array.markers.emplace_back();
+      auto & cube_marker = marker_array.markers.back();
+      cube_marker = default_cube_marker(1.0, 1.0, color);
+      cube_marker.pose = pose;
+    };
+    insert_cube_marker(info.current_obj_pose);
+  }
+  return marker_array;
+}
+
+MarkerArray showPredictedPath(const CollisionCheckDebugMap & obj_debug_vec, std::string && ns)
+{
+  int32_t id{0};
+  const auto current_time{rclcpp::Clock{RCL_ROS_TIME}.now()};
+  const auto arrow_marker_scale = createMarkerScale(1.0, 0.3, 0.3);
+  const auto default_arrow_marker = [&](const auto & color) {
+    return createDefaultMarker(
+      "map", current_time, ns, ++id, Marker::ARROW, arrow_marker_scale, color);
+  };
+
+  MarkerArray marker_array;
+  marker_array.markers.reserve(std::accumulate(
+    obj_debug_vec.cbegin(), obj_debug_vec.cend(), 0UL,
+    [&](const auto current_sum, const auto & obj_debug) {
+      const auto & [uuid, info] = obj_debug;
+      return current_sum + info.ego_predicted_path.size() + info.obj_predicted_path.size() + 2;
+    }));
+
+  for (const auto & [uuid, info] : obj_debug_vec) {
+    const auto insert_marker = [&](const auto & path, const auto & color) {
+      for (const auto & pose : path) {
+        marker_array.markers.emplace_back();
+        auto & marker = marker_array.markers.back();
+        marker = default_arrow_marker(color);
+        marker.pose = pose.pose;
+      }
+    };
+
+    insert_marker(info.ego_predicted_path, colors::aqua());
+    insert_marker(info.obj_predicted_path, colors::yellow());
+    const auto insert_expected_pose_marker = [&](const auto & pose, const auto & color) {
+      // instead of checking for distance, inserting a new marker might be more efficient
+      marker_array.markers.emplace_back();
+      auto & marker = marker_array.markers.back();
+      marker = default_arrow_marker(color);
+      marker.pose = pose;
+      marker.pose.position.z += 0.05;
```

Comment:
@zulfaqar-azmi-t4 
why 0.05 is added?

Reply:
@kyoichi-sugahara 
because the expected pose marker (red arrow) will overlapped with the predicted paths.
so to ensure it didnt overlap, the arrow is lifted on the z axis.